x-pack/metricbeat/module/cloudfoundry/container/container.go

```diff
@@ -41,5 +42,19 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 
 // Run method provides the module with a reporter with which events can be reported.
 func (m *MetricSet) Run(reporter mb.PushReporterV2) {
-	m.mod.RunContainerReporter(reporter)
+	m.mod.RunContainerReporter(&containerReporter{reporter})
+}
+
+type containerReporter struct {
+	mb.PushReporterV2
+}
+
+func (r *containerReporter) Event(event mb.Event) bool {
+	cpuPctKey := "cloudfoundry.container.cpu.pct"
+	if v, err := event.RootFields.GetValue(cpuPctKey); err == nil {
+		if v, ok := v.(float64); ok && (math.IsNaN(v) || math.IsInf(v, 0)) {
+			event.RootFields.Delete(cpuPctKey)
+		}
+	}
```

Comment:
Should we check if `err != nil && err != ErrKeyNotFound`?

Reply:
What would we do in this case? I only want to remove the field if the key is found (`err == nil`), and the value is a float with non-numeric value.
If there is any error getting the value, or it is not a float, it is ignored. (This shouldn't ever happen, but if it happens, still the value wouldn't be producing problems).

Reply:
I guess my concern is that if we're expecting there to be a value for this key but it's not there, that's unexpected behavior so something's wrong and maybe we should know about it? If you feel that's too strict, I'm okay leaving this as-is.

Reply:
Ok, I have refactored the code to log at the debug level the errors found getting the values.
I have also moved the code for the check so it is shared between both metricsets.
Let me know what you think.